compose/aspnet-mssql-compose.md

```diff
@@ -0,0 +1,142 @@
+
+# Create the Docker Compose application
+
+1. Create an empty directory named `/app` under the root and navigate into it.
+
+    This directory will be the context of your docker-compose project. For [Docker for Windows](https://docs.docker.com/docker-for-windows/#/shared-drives) and [Docker for Mac](https://docs.docker.com/docker-for-mac/#/file-sharing), you have to set up file sharing for the volume that you need to map. 
+
+1. Within your directory, use the `aspnetcore-build` Docker image to generate a sample web application within the container and into your host machine under `/app`:
+
+    ```bash
+    docker run -v /app:/app microsoft/aspnetcore-build:lts /bin/bash -c "cd app && dotnet new -t web"
+    ``` 
+
+    >**Note**: If running in Docker for Windows, make sure to add the necessary `C:/` prefix to the host mapped folder.
+
+1. Create a `Dockerfile` within the `/app` directory and add the following content:
+
+    ```
+    FROM microsoft/aspnetcore-build:lts
+
+    COPY . /app
+
+    WORKDIR /app
+
+    RUN ["dotnet", "restore"]
+
+    RUN ["dotnet", "build"]
+
+    EXPOSE 80/tcp
+
+    RUN chmod +x ./entrypoint.sh
+
+    CMD /bin/bash ./entrypoint.sh
+    ```
+
+    This file defines how to build the web app image. It will use the [microsoft/dotnet](https://hub.docker.com/r/microsoft/dotnet/), map the volume with the generated code, restore the dependencies, build the project and expose port 80. After that, it will call an `entrypoint` script that we will create in the next step. 
+
+1. The previous `Dockerfile` makes use of an entrypoint to your webapp Docker image. Create this script in a file called `entrypoint.sh` and paste the contents below.
+
+    >**Note**: Make sure to use UNIX line delimiters. The script won't work if you use Windows-based delimiters (Carriage return and line feed).
+
+    ```bash
+    #!/bin/bash
+
+    set -e
+    run_cmd="dotnet run --server.urls http://*:80"
+
+    until dotnet ef database update; do
+    >&2 echo "SQL Server is starting up"
+    sleep 1
+    done
+
+    >&2 echo "SQL Server is up - executing command"
+    exec $run_cmd
+    ```
+
+    This script will restore the database after it starts up, and then will run the application. This allows some time for the SQL Server database image to start up.
+
+1. Create a `docker-compose.yml` file. Write the following in the file, and make sure to replace the password in the `SA_PASSWORD` environment variable under `db` below. This file will define the way the images will interact as micro-services. 
+
+    >**Note**: SQL Server requires a secure password (Minimum length 8 characters, including uppercase and lowercase letters, base 10 digits and/or non-alphanumeric symbols).
```

Comment:
That is actually a bug. The container should fail to start if a strong enough password isn't specified. We're fixing that. Thanks!